ddmrp/models/stock_picking.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import models
+
+
+class StockPickgin(models.Model):
+    _inherit = "stock.picking"
+
+    def action_stock_buffer_open(self):
+        """Open a sock.buffer list related to products of the stock.picking."""
+        self.ensure_one()
+        domain = [
+            ("product_id", "in", self.mapped("move_lines.product_id.id")),
+            ("warehouse_id", "=", self.picking_type_id.warehouse_id.id),
+            ("company_id", "=", self.company_id.id),
+        ]
+        return {
+            "type": "ir.actions.act_window",
+            "name": "Stock Buffers",
+            "res_model": "stock.buffer",
+            "view_mode": "tree,form",
+            "domain": domain,
+            "context": {"search_default_procure_recommended": 1},
+        }
```

Comment:
You could reuse the existing action https://github.com/OCA/ddmrp/blob/13.0/ddmrp/views/stock_buffer_view.xml#L403 (no duplication, translations already done)
```suggestion
        action = self.env.ref("ddmrp.action_stock_buffer").read()[0]
        action["domain"] = domain
        action["context"] = {"search_default_procure_recommended": 1}
        return action
```

Reply:
@sebalix Thanks, I updated the solution.